spark/src/main/java/org/apache/iceberg/actions/BaseSparkAction.java

```diff
@@ -50,6 +55,7 @@
 
   private final SparkSession spark;
   private final JavaSparkContext sparkContext;
+  private AtomicInteger counter = new AtomicInteger();
```

Comment:
I think this can be final. I'd also call it `jobCounter` or something to be specific.

If we decide to keep the job counter in `BaseSparkAction`, then we can offer the following method:

```
  protected JobGroupInfo newJobGroupInfo(String groupId, String desc) {
    return new JobGroupInfo(groupId, desc + "-" + jobCounter.incrementAndGet(), false);
  }
```

That way, we hide the complexity of assigning the job count from individual actions.

Reply:
Few actions like `Spark3CreateAction`, `Spark3MigrateAction` implement `org.apache.iceberg.actions.Spark3CreateAction` and dont implement `org.apache.iceberg.actions.BaseSparkAction`.
Hence added the counter to a common location. We cann add the counter and its getters to both `Spark3CreateAction` and `BaseSparkAction`, what do u think @aokolnychyi

Reply:
I think we can skip action that don't extend `BaseSparkAction` for now. I am in the process of fixing the hierarchy and we can add this logic to other actions once they extend `BaseSparkAction`.